src/v/security/config_rcl.cc

```diff
@@ -183,5 +186,31 @@ result<parsed_url> parse_url(std::string_view url_view) {
     return result;
 }
 
+result<principal_mapping_rule>
+parse_principal_mapping_rule(std::string_view mapping) {
+    if (!mapping.starts_with("$.")) {
+        return errc::invalid_principal_mapping;
+    }
+    auto slash = mapping.find('/');
+    auto path = ss::sstring(mapping.substr(1, slash - 1));
+    std::replace(path.begin(), path.end(), '.', '/');
```

Comment:
Good spot. The beauty of the `substr` function is that it just does the right thing in this case.

* If slash is not found, we get `npos` which is `int_max`, so subtracting 1 is ok
* If slash is found, and because we've already checked for `starts_with` above, we can subtract 1 without concern for `out_of_bounds`.

Reply:
> If slash is not found, we get npos which is int_max, so subtracting 1 is ok
I thought `npos` was `-1`?

Reply:
> > If slash is not found, we get npos which is int_max, so subtracting 1 is ok
> 
> I thought `npos` was `-1`?

Correct. But `size_type` is unsigned.